### ISSUE TITLE
Improve CollisionObject2D class doc

### ISSUE THREAD

**Issue by udit** [godotengine/godot-docs#3204]
**Issue with current doc**: `CollisionShape2D` class doc suggests to use `CollisionObject2D` APIs to handle collisions as `CollisionShape2D` node is only a editor helper node, but it lacks proper documentation like:
- what is a shape owner
- what is the purpose of shape owner as an intermediate between Object and it's collision shape
- what is a shape_id and shape_index and how both are different

As a result, when people come up with "How to create collision shapes from script" on Github and Godot Q&A, answers still revolve around using `CollisionShape2D`

**What should be added to doc**: The doc should contain the following visualization of how collision shapes are organized in an object along with answers to above questions.

```
    PhysicsObject
         |
         -ShapeOwner0 -> Shape2D0,   Shape2D1......Shape2DN
         |               Index(0)    Index(1)      Index(N)
         |
         -ShapeOwner1 -> Shape2D0,   Shape2D1......Shape2DM
         |              Index(N+1)  Index(N+2)    Index(N+M+1)
         |
         -ShapeOwner2 -> ......
         |               
         .
         .
```
Note: Above visualization is my understanding from experimenting with the `shape_owner*` APIs, please correct if wrong.

  [1]: https://docs.godotengine.org/en/3.2/classes/class_collisionobject2d.html

**Comment by dweremeichik** [comment]
I agree that this is super confusing. I found through trial and error that you can do the following:

- Get a shape owner from a physics body (static2d, kinematic2d, etc.) by using `get_shape_owners()` any shapes attached to the physics body will be associated with of the owner ids that it returns.
- If you already have the `shape_id` you can find the owner (because there can be more than one associated with a physics body) with `shape_find_owner` using the shape_id.
- Finally, you can get a shape with `shape_owner_get_shape` once you already have the owner_id and shape_id.

A brief example of how I did it starting with a ray cast:

```gdscript
var result = space_state.intersect_ray(from, to)
	if result:
		# These are not static functions!!!
		var shape_id = result.shape
		var owner_id= result.collider.shape_find_owner(shape_id)
		var shape_2 = result.collider.shape_owner_get_shape(owner_id, shape_id)
```

**Comment by dweremeichik** [comment]
Additionally, documentation that deals with raycasts such as `Physics2DDirectSpaceState` refer to the `shape_id` as a shapes index which is potentially confusing.

**Comment by dweremeichik** [comment]
Another inconsistency that I have discovered here is that if you add a `CollisionShape2D` to a Group, you can access it and the shape any time you want. In testing, I was using this to build a navigation mesh on the fly - so it can definitely be a useful feature. It even worked in a non-debug export.
